go/src/koding/klient/uploader/uploader.go

```diff
@@ -0,0 +1,275 @@
+package uploader
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"time"
+
+	"koding/kites/keygen"
+	"koding/klient/storage"
+	"koding/logrotate"
+
+	"github.com/boltdb/bolt"
+	"github.com/koding/kite"
+	"github.com/koding/logging"
+)
+
+var defaultLog = logging.NewCustom("uploader", false)
+
+// Options represents arguments required to create a Uploader value.
+type Options struct {
+	KeygenURL string      // required
+	Kite      *kite.Kite  // required
+	Bucket    string      // required
+	Region    string      // required
+	DB        *bolt.DB    // optional; in-memory store if nil
+	Log       kite.Logger // optional; defaultLog if nil
+}
+
+// Logger is a wrapper for kite.Logger that implements logging.Logger.
```

Comment:
i didnt understand the requirement here, could you please explain? AFAIK kite.logger is already implements koding.logger as we can see here https://github.com/koding/kite/blob/02ecbfc2262fb6dd7be61a0ff23b7a79431a02fe/logger.go#L80